lua/neo-tree/sources/manager.lua

```diff
@@ -42,6 +42,7 @@ local function create_state(tabid, sd, winid)
   state.position = {
     is = { restorable = false },
   }
+  state.diagnostics_lookup = utils.get_diagnostic_counts()
```

Comment:
> You'll need to check the value of enable_diagnostics first and skip this step if that value is false | nil.

This is true. Will fix right away.

> I wonder if it makes sense to add a new event for state_created that sources can subscribe to and then add extra state initialization like this?

I agree. I will implement the simpler fix for this PR but if you plan to implement the `state_created` event, please reject this PR and add it with that event instead ;)

Reply:
Seems that `enable_diagnostics` is not passed to this function (since it's in `global_config`) so it seems to be quite a hassle to implement.

I believe it would be better to implement an event and subscribe to it in these places. 

I will close this PR. Sorry for bothering you :'(

#### filesystem
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/lua/neo-tree/sources/filesystem/init.lua#L357-L364

#### buffers
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/lua/neo-tree/sources/buffers/init.lua#L163-L168

#### git_status
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/lua/neo-tree/sources/git_status/init.lua#L73-L78